fix_test.py

```
@@ -0,0 +1,9 @@
+# The test waits up to 5 seconds for `batches_total` to be > 0.
+# If it doesn't happen within 5 seconds, it cancels the shutdown token, and then the test asserts `batches > 0`.
+# This is a classic test flake. CI machines can be slow, 5 seconds might not be enough for the tailer to start, read 50 lines, process them, and emit a batch.
+# Wait, why does the panic message say `crates/logfwd-runtime/src/pipeline/pipeline_tests/failpoints.rs`?
+# Ah! Look at `assert!(batches > 0, "...")` which is line 2082. The panic in CI is at `failpoints.rs:164:5`.
+# Wait... let me `grep` for "batches_total must be incremented even when transform filters all rows" to see if it's in `failpoints.rs`.
+import subprocess
+out = subprocess.check_output("grep -rn 'batches_total must be incremented' crates/logfwd-runtime/src/pipeline", shell=True).decode()
```

Ruff (0.15.11)

[error] `subprocess` call with `shell=True` seems safe, but may be changed in the future; consider rewriting without `shell`

(S602)